sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java

```diff
@@ -117,6 +119,26 @@ public String getDirectoryName() {
         return getObjectName();
     }
 
+    /**
+     * Creates a new {@link DataLakeDirectoryAsyncClient} with the specified {@code customerProvidedKey}.
+     *
+     * @param customerProvidedKey the {@link CustomerProvidedKey} for the blob,
+     * pass {@code null} to use no customer provided key.
```

Comment:
So, is null a way to get back to a non-CPK client? If this is following previous patterns I'm good with it, just a little confused on the handling.

Also, should this use `directory` instead of `blob`?

```suggestion
     * @param customerProvidedKey the {@link CustomerProvidedKey} for the directory,
     * pass {@code null} to use no customer provided key.
```

Reply:
Yes, this is following the same thing we have in [blobs](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobClientBase.java#L132)